modules/core/src/matrix_operations.cpp

```diff
@@ -605,7 +605,7 @@ namespace cv
 {
 
 template<typename T, typename ST, class Op> static void
-reduceR_( const Mat& srcmat, Mat& dstmat )
+reduceR_( const Mat& srcmat, Mat& dstmat, bool applyOpOnFirstRow = false)
```

Comment:
Implementation changes doesn't look good.
What if you want to calculate product instead of sum? "0" value doesnt' work anymore

IMHO, It is better to define separate "Init" functor:
- NoOp for old cases
- new SqValue

```.diff
-buf[i] = src[i];
+buf[i] = initOp(src[i]);
```

Reply:
Good idea, I am on it